datafusion/core/src/physical_optimizer/topk_aggregation.rs

```diff
@@ -51,7 +52,7 @@ impl TopKAggregation {
         if desc != order.options.descending {
             return None;
         }
-        let group_key = aggr.group_expr().expr().first()?;
+        let group_key = aggr.group_expr().expr().iter().exactly_one().ok()?;
```

Comment:
TIL `exactly_one`  👍  makes sense to me. 

Another way to write this that is more verbose but perhaps clearer might be:

```rust
if agg.group_expr().len() != 1 {
  return None;
}
```